addons/html_editor/static/src/utils/color.js

```diff
@@ -118,6 +118,92 @@ export function rgbToHex(rgb = "", node = null) {
     }
 }
 
+/**
+ * Converts an RGBA or RGB color string to a hexadecimal color string.
+ * - If the input color is already in hex format, it returns the hex string directly.
+ * - If the input color is in rgba format, it converts it to a hex string, including the alpha value.
+ * - If the input color is in rgb format, it converts it to a hex string (with no alpha).
+ *
+ * @param {string} rgba - The color string to convert (can be in RGBA, RGB, or hex format).
+ * @returns {string} - The resulting color in hex format (including alpha if applicable).
+ */
+export function rgbaToHex(rgba = "") {
+    if (rgba.startsWith("#")) {
+        return rgba;
+    } else if (rgba.startsWith("rgba")) {
+        const values = rgba.match(/[\d.]{1,5}/g) || [];
+        const alpha = values.length === 4 ? parseFloat(values.pop()) : 1;
+        const hexRgb = values
+            .map((value) => {
+                const hex = parseInt(value).toString(16);
+                return hex.length === 1 ? "0" + hex : hex;
+            })
+            .join("");
+        const hexAlpha = Math.round(alpha * 255).toString(16);
+        const finalAlpha = hexAlpha.length === 1 ? "0" + hexAlpha : hexAlpha;
+
+        return "#" + hexRgb + finalAlpha;
+    } else {
+        return (
+            "#" +
+            (rgba.match(/\d{1,3}/g) || [])
+                .map((x) => {
+                    x = parseInt(x).toString(16);
+                    return x.length === 1 ? "0" + x : x;
+                })
+                .join("")
+        );
+    }
+}
+
+/**
+ * Blends an RGBA color with the background color of a given DOM node.
+ * - If the input color is not RGBA, it is converted to hex.
+ * - If the node has an RGBA background, the function recursively blends it with its parent's background.
+ * - If no valid background is found, it defaults to white (#FFFFFF).
+ *
+ * @param {string} color - The RGBA color to blend.
+ * @param {HTMLElement|null} node - The DOM node to get the background color from.
+ * @returns {string} - The resulting blended color as a hex string.
+ */
+export function blendColors(color, node) {
+    if (!color.startsWith("rgba")) {
+        return rgbaToHex(color);
+    }
+    let bgRgbValues = [255, 255, 255];
+    if (node) {
+        let bgColor = getComputedStyle(node).backgroundColor;
+
+        if (bgColor.startsWith("rgba")) {
```

Comment:
This will always be true since you already check its opposite and return on the first line.

Reply:
The condition in the first line applies to both text color and background color. However, inside, I am only checking for the background color. So, if the text color is `rgba` but the background color is not, this condition will be false.